tutorials/tree/dictionary/CMakeLists.txt

```diff
@@ -0,0 +1,55 @@
+#####################################################################################################################
+
+# Jorge Agramunt Ros    @ IFIC(Valencia,Spain)  jorge.agramunt@ific.uv.es
+# Alvaro Tolosa Delgado @ IFIC(Valencia,Spain)  alvaro.tolosa@ific.uv.es
+# Copyright (c) 2016 Jorge Agramunt & Alvaro Tolosa. All rights reserved.
+
+#####################################################################################################################
+
+# Details about integrating ROOT into CMake projects:
+#     https://root.cern/manual/integrate_root_into_my_cmake_project/
+
+#####################################################################################################################
+
+# CMakeLists.txt for event package. It creates a library with dictionary and a main program
+cmake_minimum_required(VERSION 3.9 FATAL_ERROR)
+
+project(treeUsingCustomClass)
+
+# You need to tell CMake where to find the ROOT installation. This can be done in a number of ways:
+#   - ROOT built with classic configure/make use the provided $ROOTSYS/etc/cmake/FindROOT.cmake
+#   - ROOT built with CMake. Add in CMAKE_PREFIX_PATH the installation prefix for ROOT
+list(APPEND CMAKE_PREFIX_PATH $ENV{ROOTSYS})
+
+#---Locate the ROOT package and defines a number of variables (e.g. ROOT_INCLUDE_DIRS)
+find_package(ROOT REQUIRED COMPONENTS MathCore RIO Hist Tree Net Graf3d Graf Gpad Thread TreePlayer)
+
+#---Define useful ROOT functions and macros (e.g. ROOT_GENERATE_DICTIONARY)
+include(${ROOT_USE_FILE})
+
+#---Add include directory of ROOT to the build
+include_directories(${CMAKE_SOURCE_DIR} ${ROOT_INCLUDE_DIRS})
+
+#---Add library directory of ROOT to the build
+link_directories(${ROOT_LIBRARY_DIR})
+
+#---Set Flags
+add_definitions(${ROOT_CXX_FLAGS}  )
```

Comment:
These should not be needed .. They are meant to be inherited/transferred automatically after adding the `ROOT` libraries as dependencies of a given target.